crates/orderbook/src/orderbook.rs

```diff
@@ -277,4 +355,72 @@ mod tests {
             .unwrap();
         assert_eq!(result, &orders[1..2]);
     }
+
+    #[test]
+    fn computes_u256_prices_normalized_to_1e18() {
+        assert_eq!(
+            to_normalized_price(0.5),
+            U256::from(500_000_000_000_000_000_u128)
+        );
+    }
+
+    #[tokio::test]
+    async fn filters_tokens_without_native_prices() {
+        let token1 = H160([1; 20]);
+        let token2 = H160([2; 20]);
+        let token3 = H160([3; 20]);
+
+        let orders = vec![
+            OrderBuilder::default()
+                .with_sell_token(token1)
+                .with_buy_token(token2)
+                .build(),
+            OrderBuilder::default()
+                .with_sell_token(token2)
+                .with_buy_token(token3)
+                .build(),
+            OrderBuilder::default()
+                .with_sell_token(token1)
+                .with_buy_token(token3)
+                .build(),
+        ];
+        let prices = btreemap! {
+            token1 => 0.5,
+            token3 => 2.0,
+        };
+
+        let mut native_price_estimator = MockNativePriceEstimating::new();
+        native_price_estimator
+            .expect_estimate_native_prices()
+            // deal with undeterministic ordering of `HashSet`.
+            .withf(move |tokens| {
+                tokens.iter().cloned().collect::<HashSet<_>>() == hashset!(token1, token2, token3)
+            })
```

Comment:
This is a little unfortunate but changing the implementation of `filter_tokens_without_native_prices` so you can compare against `&[H160]` doesn't seem worth it.

Reply:
Its just a unit test so I'm not too worried about optimizing it.

The other option is to include a:
```rust
#[cfg(test)]
let traded_tokens = {
    let mut tokens = traded_tokens;
    tokens.sort_unstable();
    tokens
};
```

In the implementation. I slightly prefer making the mock code a little dirtier than sorting the token list when building in `test`, but I have no strong feelings either way.

Reply:
I agree with you and I don't see a need to change anything here.
This was just something that stood out to me.